### DISCUSSION TITLE
Renaming the BlueBuild binary file

### DISCUSSION THREAD

**Discussion by gmpinder** [blue-build/.github#7] [Polls]
@bayou-brogrammer , @xynydev , @fiftydinar , @gerblesh 

In our Discord, there was some talk about possibly renaming the binary file for the CLI tool from `bb` to `bluebuild`. Now this change would be a fairly disruptive one to make as we would need to make sure to update README's, website docs, the Github Action, and updating the cli tool itself. There was also the suggestion of allowing both for an easier transition. 

Since this would be a decent sized change we should vote on how we would want to go about this, so I've created this poll. Discussions about this is encouraged!

**Comment by fiftydinar** [comment]
I vote for both.

However, I would emphasize on using `bluebuild` primarily in docs & other mentioned stuff for having name consistency, while I would casually mention in some appropriate places that `bb` can be used as the alias to `bluebuild`.

**Reply by gmpinder** [reply]
Yeah and I'm thinking if we do both, we should probably use a symlink so that we aren't shipping 2 actual binaries that could get out of sync. So in my suggestion, `bluebuild` would be the actual binary so that it helps prevent being overwritten by some other tool that the user installed.

**Reply by fiftydinar** [reply]
That sounds good, I agree

**Reply by gerblesh** [reply]
I'm thinking of axing `bb` mainly due to the early nature of the project. While `bb` is nice as a shorthand I think it's far too generic as a command in the CLI and could conflict with other programs. `bluebuild` makes more sense in this case. Axing `bb` is an option because the project is early enough in development to where nobody is actually using it besides a couple people dogfooding, so it would make sense to break compat early rather than later (don't want an `egrep` and `fgrep` type situation)

**Reply by gmpinder** [reply]
That's actually a really good point. It would seem that my vote cannot be changed, so I want to say that I choose `bluebuild` only.

**Reply by fiftydinar** [reply]
I initially wanted `bluebuild` only, but gmpinder explained to me that he makes `bb` command the 1st preference compared to other potential `bb`'s used by other programs (with path preference).

However, even when I voted for `bluebuild` + `bb`, I noted a strong preference for `bluebuild`.

So in this case, you can count me in for `bluebuild` only too, as you mentioned `egrep` vs `fgrep` situation, which we want to avoid.

**Comment by xynydev** [comment]
I'm ambivalent on this issue. As a more aesthetics-oriented person I like the sound of `bb bugreport` and `bb build` more than `bluebuild bugreport`  and `bluebuild build`. There seems to be a pretty strong concensus in calling it `bluebuild`, though, as @gmpinder expressed a swayed opinion and @fiftydinar was originally swayed by him. I'll side with the popular opinion here.

**Comment by gmpinder** [comment]
Work has already started on this! We have the binary rename merged in. We'll want to coordinate for the next release of the CLI so that docs and actions are updated.
